src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -2957,6 +2959,11 @@ def forward(
         output_hidden_states = (
             output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
         )
+        if self.config.use_weighted_layer_sum:
```

Comment:
Following the convention we use in Wav2Vec2 by forcing `output_hidden_states=True` when `use_weighted_layer_sum=True`: 
https://github.com/huggingface/transformers/blob/2c1eebc1216549d8195d7d1c6adb8b99afee3ec5/src/transformers/models/wav2vec2/modeling_wav2vec2.py#L2093